### ISSUE TITLE
A = CR Factorization

### ISSUE THREAD

**Issue by prairie-guy** [blegat/RowEchelon.jl#18]
I’ve read through the debate about adding [rref](https://github.com/JuliaLang/julia/pull/9804) to `Base`. One of the arguments for keeping it around was pedagogical, as it is instructive in teaching Linear Algebra. I appreciate it finding a home here in `RowEchelon.jl`, as I am taking several of Gilbert Strang’s MIT Linear Algebra classes including [18.065](https://ocw.mit.edu/courses/18-065-matrix-methods-in-data-analysis-signal-processing-and-machine-learning-spring-2018/) and his most [recent.](https://ocw.mit.edu/courses/res-18-010-a-2020-vision-of-linear-algebra-spring-2020/) In them `rref` is used extensively. 

In these classes, he has been extolling the virtues of the [A = CR](https://math.mit.edu/~gs/everyone/lucrweb.pdf) factorization. He  admits it is not intended for numerical computation, but that it is useful for teaching purposes. It not a complicated function, but I have implemented it. Is it something that would find a home here, especially that almost all of it is based upon `rref_with_piviots`? If so, I can provide a `PR`

**Comment by blegat** [comment]
Yes, if the code is simple, that could be a good fit for this package
